src/main/scala/cromwell/util/EnhancedExponentialBackoff.scala

```diff
@@ -0,0 +1,54 @@
+package cromwell.util
+
+import com.google.api.client.util.ExponentialBackOff
+
+import scala.concurrent.duration.{Duration, FiniteDuration}
+
+sealed trait Backoff {
+  def backoffMillis: Long
+  def next: InitializedBackoff
+}
+
+/**
+  * Useful when using Backoffs in a "functional" backoff.next.backoffMillis way.
+  * This prevents the first call to next to shunt out the first value of backoffMillis.
+  */
+final case class UnInitializedBackoff(next: InitializedBackoff) extends Backoff {
+  override def backoffMillis = throw new UninitializedError()
+}
+
+sealed trait InitializedBackoff extends Backoff {
+  def uninitialized: UnInitializedBackoff = UnInitializedBackoff(this)
+}
+
+final class InitialGapBackoff(initialGapMillis: FiniteDuration, googleBackoff: ExponentialBackOff) extends InitializedBackoff {
+
+  assert(initialGapMillis.compareTo(Duration.Zero) != 0, "Initial gap cannot be null, use SimpleBackoff instead.")
```

Comment:
The common sense would be that if you give a initial gap = 0, you would get the same thing as if you used a SimpleBackoff (ie the first interval you get would be initialIntervalMillis).
But that is not the case because you will get 0.
To be sure that this is never used inappropriately an assert seemed the right thing - plus this is more destined to the developer than the users.